Add parse XML docs (#6822)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/parse-xml.md

---
layout: default
title: parse_xml
parent: Processors
grand_parent: Pipelines
nav_order: 83
---

# parse_xml

The `parse_xml` processor parses XML data for an event.

## Configuration

You can configure the `parse_xml` processor with the following options.

| Option | Required | Type | Description |
| :--- | :--- | :--- | :--- | 
| `source` | No | String | The field in the `event` that is parsed. Default value is `message`. |
| `destination` | No | String | The destination field of the parsed XML. Defaults to the root of the `event`. Cannot be `""`, `/`, or any white-space-only `string` because these are not valid `event` fields. |
| `pointer` | No | String | A JSON pointer to the field to be parsed. There is no `pointer` by default, meaning that the entire `source` is parsed. The `pointer` can access JSON array indexes as well. If the JSON pointer is invalid, then the entire `source` data is parsed into the outgoing `event`. If the key that is pointed to already exists in the `event` and the `destination` is the root, then the pointer uses the entire path of the key. |

> | `tags_on_failure` | No | String | A list of strings that specify the tags to be set in the event that the processors fails or an unknown exception occurs while parsing. 

Reviewer: This looks good! Only thing missing is the `parse_when` attribute to conditionally run the processor. This is like drop_when (https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/drop-events/) or grok_when, just for this processor.

Author: @graytaylor0 added, thanks

---
layout: default
title: parse_xml
parent: Processors
grand_parent: Pipelines
nav_order: 83
---

# parse_xml

The `parse_xml` processor parses XML data for an event.

## Configuration

You can configure the `parse_xml` processor with the following options.

| Option | Required | Type | Description |
| :--- | :--- | :--- | :--- | 

> | `source` | No | String | The field in the `event` that is parsed. Default value is `message`. |

Reviewer 2: In the `event` object?

> | `source` | No | String | The field in the `event` object that is parsed. Default value is `message`. |

Reviewer 3: As written, it's not clear whether the field or the object is parsed.

> | `source` | No | String | The field in the `event` object that is parsed. Default value is `message`. |

Reviewer 4: ```suggestion
| `source` | No | String | Tells the processor which field contained in the `event` to parse. Default value is `message`. |
```

Reviewer 4: @natebower: Does this work?

Reviewer 3: Can it just be "Specifies which `event` field to parse"?

> | `destination` | No | String | The destination field of the parsed XML. Defaults to the root of the `event`. Cannot be `""`, `/`, or any white-space-only `string` because these are not valid `event` fields. |

Reviewer 2: ```suggestion
| `destination` | No | String | The destination field of the parsed XML. Defaults to the root of the `event`. Cannot be `""`, `/`, or any white-space-only string because these are not valid `event` fields. |
```

> | `pointer` | No | String | A JSON pointer to the field to be parsed. There is no `pointer` by default, meaning that the entire `source` is parsed. The `pointer` can access JSON array indexes as well. If the JSON pointer is invalid, then the entire `source` data is parsed into the outgoing `event`. If the key that is pointed to already exists in the `event` and the `destination` is the root, then the pointer uses the entire path of the key. |

Reviewer 2: Does "no pointer by default" mean "the default pointer is empty? Null?

Author: Null, fixed.

> | `parse_when` | No | String | Specifies under what conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper Expression Syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |

Reviewer 3: ```suggestion
| `parse_when` | No | String | Specifies under what conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper expression syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |
```

> | `parse_when` | No | String | Specifies under what conditions the processor should perform parsing. Default is no condition. Accepts a Data Prepper expression string following the [Data Prepper Expression Syntax]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/expression-syntax/). |

Reviewer 3: "Following" can be read as either "after" or "in accordance with" here.

> | `tags_on_failure` | No | String | A list of strings that specify the tags to be set in the event that the processors fails or an unknown exception occurs while parsing. 

Reviewer 2: ```suggestion
| `tags_on_failure` | No | String | A list of strings that specify the tags to be set if the processor fails or an unknown exception occurs while parsing. 
```
